### PR TITLE
ADD: auth context and signup login functions

### PR DESCRIPTION
**Pull request by AnubhavChakrabortynits**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+VITE_SIGNUP_URL="http://localhost:5000/api/v1/auth/signup"
+VITE_LOGIN_URL="http://localhost:5000/api/v1/auth/login"
```
Comment (codadept) [inline .env]: Don't use individual routes in environment variables.

Just add `VITE_BASE_URL="http://localhost:5000/api/v1"`

and use it in js as 

```js
`${import.meta.env.VITE_BASE_URL}/auth/login`
```

**Review by codadept — CHANGES_REQUESTED** [review]
Move your login/signup code to the auth context itself and handle it there and in the Login/Signup component just call the corresponding context functions. 

Change your code accordingly

**Comment by codadept** [comment]
Also fix the merge conflicts

**Comment by AnubhavChakrabortynits** [comment]
made changes as requested

File: src/Global/Auth/authContext.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+
+const UserContext = createContext();
```
Comment (codadept) [inline src/Global/Auth/authContext.js]: Pass the default value in `React.createContext`

**Review by codadept — APPROVED** [review]

**Comment by AnubhavChakrabortynits** [comment]
deafult value provided

File: src/Global/Auth/authContext.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+
+const UserContext = createContext("User");
```
Comment (codadept) [inline src/Global/Auth/authContext.js]: Default value means the object you are storing in the context. e.g.

```js

const UserContext = createContext({
    username: "",
    email: "",
    address: {
        street: "",
        city: "",
    }
});
```

Whatever it is according to your context.

**Review by codadept — CHANGES_REQUESTED** [review]

**Comment by AnubhavChakrabortynits** [comment]
Fixed default value of authContext

**Review by codadept — APPROVED** [review]

**Pull request merged**